docs/concepts/joins.rst

```diff
@@ -350,6 +350,32 @@ optimized.*
    Complex queries are broken down into subqueries that are run on their shards
    before joining.
 
+.. _join-optim-eliminate-cross-joins:
+
+Cross join elimination
+----------------------------
+
+The optimizer will try to eliminate cross joins in the query plan by changing
+the join-order. When reordering, it will try to preserve the original join order
+as much as possible. If a cross join cannot be eliminated, the original join
+order will be maintained. It will try to eliminate any cross join
+it can e.g.:
```

Comment:
Maybe add a sentence or two that defines or explains what cross join elimination is, or what it does. 

E.g. something like "Cross join elimination replaces a CROSS JOIN with a INNER JOIN if query conditions used in the WHERE clause or other join conditions allow for it. An example: ..."

The whole paragraph kinda assumes that people are already familiar with it.

Reply:
👍